### PR TITLE
Allow global meta to provide opt-out of package extension adding

### PR DESCRIPTION
**Pull request by blink1073**
Fixes https://github.com/jspm/jspm-cli/issues/1341.

### CODE REVIEW

File: lib/package.js [hunk 1]
```diff
@@ -164,6 +164,13 @@
           skipExtension = true;
       });
 
+    // exact global meta or meta with any content after the last wildcard skips extension
+    if (!skipExtension && System.modules)
```
Comment (guybedford) [inline lib/package.js]: `loader.meta`

File: lib/package.js [hunk 2]
```diff
@@ -164,6 +164,13 @@
...
+      getMetaMatches(System.modules, pkgName, subPath, function(metaPattern, matchMeta, matchDepth) {
```
Comment (guybedford) [inline lib/package.js]: `getMetaMatches` is designed for package meta, so won't work here I don't think.

**Comment by guybedford** [comment]
https://github.com/systemjs/systemjs/blob/master/lib/package.js#L470 actually doesn't use the `pkgName` argument it seems so it will work fine here.

File: lib/package.js (hunk 2 above)
Comment (guybedford) [inline lib/package.js]: `subPath` would need to be the full path.

File: lib/package.js (hunk 1 above)
Comment (blink1073) [inline lib/package.js]: Fixed.

File: lib/package.js (hunk 2 above)
Comment (blink1073) [inline lib/package.js]: This does work as is.  What do you mean by '"full path"?

File: lib/package.js [hunk 3]
```diff
@@ -164,6 +164,13 @@
           skipExtension = true;
       });
 
+    // exact global meta or meta with any content after the last wildcard skips extension
+    if (!skipExtension && loader.meta)
+      getMetaMatches(loader.meta, pkgName, subPath, function(metaPattern, matchMeta, matchDepth) {
```
Comment (guybedford) [inline lib/package.js]: `subPath` can be replaced with `pkgName + '/' + basePath + subpath`

Also the `pkgName` argument to `getMetaMatches` can be entirely removed from the signature.

Comment (blink1073) [inline lib/package.js]: Updated.

Comment (blink1073) [inline lib/package.js]: And it still works in my patched application.

**Comment by guybedford** [comment]
I will likely post out a release next week some time with this fix.

**Comment by blink1073** [comment]
Awesome, thank you very much.

**Pull request merged**